src/discord-cluster-manager/leaderboard_db.py

```diff
@@ -114,8 +114,10 @@ def create_leaderboard(self, leaderboard: LeaderboardItem):
             )
             self.connection.commit()
         except psycopg2.Error as e:
-            print(f"Error during leaderboard creation: {e}")
             self.connection.rollback()  # Ensure rollback if error occurs
+            return f"Error during leaderboard creation: {e}"
```

Comment:
I'm nervous about returning the value with the stringified exception in it. The reason is that I'm not sure what's going to be in `e`. I wouldn't expect it to include sensitive information (like the database URL), but it could. What I would suggest is to log the full string (`f"Error during leaderboard creation: {e}"`) but only return `"Error during leaderboard creation"`.

Reply:
Good point, for now I won't log it (we can change this later) but will remove error.